packages/terminal/src/node/terminal-server.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { ILogger } from '@theia/core/lib/common/logger';
+import {
+    ITerminalServer, ITerminalClient, ITerminalExitEvent,
+    ITerminalErrorEvent, ITerminalServerOptions
+} from '../common/terminal-protocol';
+import { TerminalProcess, TerminalProcessFactory } from '@theia/core/lib/node/process/terminal-process';
+import { ProcessManager } from '@theia/core/lib/node/process/process-manager';
+import URI from "@theia/core/lib/common/uri";
+
+@injectable()
+export class TerminalServer implements ITerminalServer {
+
+    /* Logger client to send notifications to.  */
+    protected client: ITerminalClient | undefined = undefined;
+
+    constructor(
+        @inject(TerminalProcessFactory) protected readonly terminalFactory: TerminalProcessFactory,
+        @inject(ProcessManager) protected readonly processManager: ProcessManager,
+        @inject(ILogger) protected readonly logger: ILogger) { }
+
+    create(options: object): Promise<number> {
+        const termOptions = <ITerminalServerOptions>options;
+        try {
+            const term = this.terminalFactory({
+                'command': termOptions.command,
+                'args': termOptions.args,
+                'options': termOptions.options
+            });
+
+            this.postCreate(term);
+
+            return Promise.resolve(term.id);
+        } catch (error) {
+            this.logger.error(`Error while creating terminal: ${error}`);
+            return Promise.resolve(-1);
+        }
+    }
+
+    protected postCreate(term: TerminalProcess) {
+        term.onError(error => {
+            this.logger.error(`Terminal pid: ${term.pid} error: ${error}, closing it.`);
+
+            if (this.client !== undefined) {
+                this.client.onTerminalError(
+                    <ITerminalErrorEvent>{
+                        'terminalId': term.id,
+                        'error': error
+                    });
+            }
+            term.dispose();
+        });
+
+        term.onExit(event => {
+            if (this.client !== undefined) {
+                this.client.onTerminalExitChanged(
+                    <ITerminalExitEvent>{
```

Comment:
i think you don't need to provide types in such places, tsc knows it from the method signature

Reply:
yeah I wasn't sure about that I'll retry

Reply:
Actually that's even a mistake since it force cast to a type, I should avoid using that I'll research more better ways in general. Just removed it in this case.